content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,137 @@
+    
+1. Distribute new CA private keys (`ca.key` and `front-proxy-ca.key`) to all your control plane nodes in the Kubernetes certificates directory.
+
+1. Update *Kubernetes controller manager's* `--root-ca-file` to include both old and new CA and restart controller manager.
+
+    Any service account created after this point will get secrets that include both old and new CAs.
+
+    {{< note >}}
+
+    Remove the flag `--client-ca-file` from the *Kubernetes controller manager* configuration. You can also replace the existing client CA file or change this configuration item to reference a new, updated CA. [Issue 1350](https://github.com/kubernetes/kubeadm/issues/1350) tracks an issue with *Kubernetes controller manager* being unable to accept a CA bundle.
+
+    {{< /note >}}
+
+1. Update all service account tokens to include both old + new CA certificates.
+    
+    If any pods started before new CA is picked by API servers, they will get this update and trust both old + new CAs.
+
+    ```shell
+    base64_encoded_ca="$(base64 <path to file containing both old + new CAs>)"
+
+    for namespace in $(kubectl get ns --no-headers | awk '{print $1}'); do
+        for token in $(kubectl get secrets --namespace "$namespace" --field-selector type=kubernetes.io/service-account-token -o name); do
+            kubectl get $token --namespace "$namespace" -o yaml | \
+            /bin/sed "s/\(ca.crt:\).*/\1 ${base64_encoded_ca}" | \
+            kubectl apply -f -
+        done
+    done
+    ```
+
+1. Restart all pods using in-cluster configs (ex: kube-proxy, coredns, etc) so they can use the updated certificate authority data from *ServiceAccount* secrets.
+
+    *   Make sure coredns, kube-proxy and other pods using in-cluster configs are working as expected.
+
+1. Append new CA to the file against `--client-ca-file` flag in the `kube-apiserver` and `kube-scheduler` configuration.
+
+    {{< note >}}
+
+    If you used `kubeadm` to deploy your cluster, the control plane components run as {{< glossary_tooltip text="static Pods" term_id="static-pod" >}}, managed by the {{< glossary_tooltip text="kubelet" term_id="kubelet" >}}.
+
+    {{< /note >}}
+
+1. Update certificates for user accounts by replacing the content of `client-certificate-data` and `client-key-data` respectively and the `certificate-authority-data` section in all `admin.conf`, `controller-manager.conf`, `scheduler.conf` files with Base64-encoded new certificate data:
+
+    For information about creating certificates for individual user accounts, see [Configure certificates for user accounts](/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts).
+
+1. Restart API servers with the certificates (`apiserver.crt`, `apiserver-kubelet-client.crt`, and `front-proxy-client.crt`) signed by new CA. You can use the existing private keys or new private keys. If you changed the private keys then update these in the Kubernetes certificates directory as well.
```

Comment:
This step comes too soon, kubelets and aggregate API servers aren't configured to trust the new CA yet.

Can you clarify that this should be performed in a rolling fashion?

Reply:
Yes, thats better. Will make that change.

Reply:
This is still too soon, you need to ensure the kubelets and aggregate APIs trust the new CAs before using a new API server certificate